src/main/java/mods/railcraft/common/blocks/detector/types/DetectorRouting.java

```diff
@@ -105,8 +105,9 @@ public int testCarts(List<EntityMinecart> carts) {
             if (routingController.getButtonState() == RoutingButtonState.PRIVATE)
                 if (!getOwner().equals(CartToolsAPI.getCartOwner(cart)))
                     continue;
-            if (logic.matches(this, cart))
-                return FULL_POWER;
+            int value = logic.evaluate(this, cart);
+            if (value != 0)
```

Comment:
This preserves the original behavior of moving to the next cart in `carts` if the table does not match (plus we don't have to reevaluate on the following line). Though I see now that the result depends on the order of `carts` if multiple carts trigger different nonzero signal levels. We could take the maximum value (or something like that) in order to make it consistent.